hooks/pre/10-reset_data.rb

```diff
@@ -2,22 +2,22 @@
 require 'English'
 
 def reset
-  if foreman_installed?
+  if foreman_maintain_present?
 
     stop_services
```

Comment:
This relies on foreman-maintain and the installer can't depend on that. It simply doesn't exist on Debian. That's a regression for the regular (non-katello) installer users.

Reply:
@kgaikwad We really need foreman-maintain on Debian to prevent this kind of incongruence within our implementation.

Reply:
I'd strongly prefer to get rid of the foreman-maintain dependency in `stop_services` altogether. We can simply come up with a list of services and replace it all with:
```
def stop_services(services)
  running = services.filter do |service|
    system('systemctl', 'is-active', service, in: '/dev/null', out: '/dev/null', err: '/dev/null')
  end
  system('systemctl', 'stop', *services, in: '/dev/null', out: '/dev/null', err: '/dev/null')
end
```

All the priority things that foreman-maintain does is IMHO not not needed and systemd can figure it out. It does so on system shutdown and there it's not a problem.

Reply:
@ehelms, 
I understand your concern. Though we resumed our work on upstream support for upgrade scenarios however we are currently focusing on rhel & centos support first. So, it may take little more time to have Debian support.

Reply:
I'm preparing a PR that removes the need for it, writing tests now.

Reply:
https://github.com/theforeman/foreman-installer/pull/510 needs polishing and testing.